engine/src/graphics/pipeline.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023-present Genesis Engine contributors (see LICENSE.txt)
+
+#include "graphics/pipeline.hpp"
+#include "graphics/vkHelpers.hpp"
+
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace gen
+{
+	GraphicsPipeline::GraphicsPipeline(GraphicsDevice & device, const PipelineConfigInfo & configInfo)
+		: m_device{device}, m_configInfo{configInfo}
+	{
+		createGraphicsPipeline();
+		m_logger.info("Graphics pipeline constructed");
+	}
+
+	GraphicsPipeline::~GraphicsPipeline()
+	{
+		m_logger.info("Graphics pipeline destructed");
+	}
+
+	PipelineConfigInfo GraphicsPipeline::defaultPipelineConfigInfo(const mim::vec2i& extent)
```

Comment:
The initialization / setting of some stuff in this function is redundant as it is the default anyway. Understandable if you want to be as explicit as possible, but just letting you know that it adds boilerplate.